### PR TITLE
Modularization Enhancements

### PR DESCRIPTION
**Pull request by dansiegel**
Enhancements from issue #1139

### CODE REVIEW

File: Source/Xamarin/Prism.Autofac.Forms/PrismApplication.cs [hunk 1]
```diff
@@ -119,7 +132,14 @@ protected override ContainerBuilder CreateContainer()
         /// </summary>
         /// <returns></returns>
         protected override IModuleManager CreateModuleManager() =>
-            Container.Resolve<IModuleManager>();
+            new AutofacModuleManager(ModuleInitializer, ModuleCatalog);
```
Comment (bartlannoeye) [inline Source/Xamarin/Prism.Autofac.Forms/PrismApplication.cs]: personally I would keep these accessors on a single line.

**Comment by brianlagunas** [comment]
I'm really not feeling the addition of the AutofacModuleBase class.  I also don't like the fact the the IModule doesn't work.  IModule should work regdarless of if they are using the IAutofacModule or not.

**Comment by dansiegel** [comment]
Unless the module can access the ContainerBuilder there is no way for the module to initialize it. Based on the way the AutofacModuleInitializer works with the Activator because we can't call `Builder.Build()`, it really requires that the Module doesn't have any parameters in the ctor which makes the IModule.Initialize useless.

**Comment by brianlagunas** [comment]
pull out the autofac changes and submit those as a separate PR.

**Comment by brianlagunas** [comment]
I moved your changes into another PR and I am closing this one.

**Pull request closed without merging**